opendevin/browser_server/browser_process.py

```diff
@@ -0,0 +1,27 @@
+from multiprocessing.connection import Connection
+
+import browsergym.core  # noqa F401 (we register the openended task as a gym environment)
+import gymnasium as gym
+
+from opendevin.browser_server.utils import image_to_png_base64_url
+
+
+def start_browser(conn: Connection):
```

Comment:
Considering abstract this into a `Browser` class that has:
- `start()` logic that manages the `Process`
- `image_to_png_base64_url` method (so we don't need a separate `utils.py` file)
- `step(action_str)` that handles communication to the process via Pipe & error messages, so that people don't need to know what happens under the hood of the `Browser` class. They can simply interact with `controller.browser`.

Then, in each browser action, you can simply do something like this. You might need to handle the observation, but you get the idea!

```
@dataclass
class BrowseURLAction(ExecutableAction):
    url: str
    action: str = ActionType.BROWSE
    async def run(self, controller: 'AgentController') -> BrowserOutputObservation:  # type: ignore
        asked_url = self.url
        if not asked_url.startswith('http'):
            asked_url = os.path.abspath(os.curdir) + self.url
        return controller.browser.step(f'goto("{asked_url}")')
```

Reply:
This is a good comment!